server/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregationBuilder.java

```diff
@@ -678,17 +678,18 @@ public static TopHitsAggregationBuilder parse(String aggregationName, XContentPa
                                 if (token == XContentParser.Token.FIELD_NAME) {
                                     currentFieldName = parser.currentName();
                                 } else if (token.isValue()) {
-                                    if (SearchSourceBuilder.SCRIPT_FIELD.match(currentFieldName)) {
+                                    if (SearchSourceBuilder.SCRIPT_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
                                         script = Script.parse(parser);
-                                    } else if (SearchSourceBuilder.IGNORE_FAILURE_FIELD.match(currentFieldName)) {
+                                    } else if (SearchSourceBuilder.IGNORE_FAILURE_FIELD.match(currentFieldName,
+                                        parser.getDeprecationHandler())) {
```

Comment:
Just like last time, would you mind indenting this one more level?

Reply:
If you think we shouldn't I'm ok with it but I find it so much more readable that way.